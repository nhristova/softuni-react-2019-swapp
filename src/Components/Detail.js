import React from 'react';
export const Detail = props => {
  return (
    <div>
      <span className="text-highlight-2">{props.type}: </span>
      <span className="text-highlight">{props.value}</span>
    </div>
  );
};
