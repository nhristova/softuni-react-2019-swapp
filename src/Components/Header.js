import React from 'react';
import { Link } from 'react-router-dom';
import { useApolloClient } from '@apollo/react-hooks';

import { ReactComponent as SignOut } from '../assets/sign-out.svg';
export const Header = props => {
  const client = useApolloClient();

  const handleSignOut = () => {
    localStorage.removeItem('token');
    client.writeData({ data: { authenticated: false } });
  };

  return (
    <header className="page-header">
      <div className="header-logo" onClick={props.changeTheme}>
        SWAPP
      </div>
      <nav className="header-navigation">
        <Link to="/episodes">Episodes</Link>
        <Link to="/characters">Characters</Link>
        <SignOut
          viewBox="13 15 70 70"
          className="logout-icon"
          onClick={handleSignOut}
        />
      </nav>
    </header>
  );
};
