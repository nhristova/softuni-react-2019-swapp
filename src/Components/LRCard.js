import React from 'react';
import { Link } from 'react-router-dom';

import noImage from '../assets/no-image-darth.png';

export const LRCard = props => {
  const item = props.item;
  const image = item.image || noImage;
  return (
    <div className={`lr-card ${props.size}`}>
      <img className="lr-card-photo" src={image} alt={item.name} />
      <div className="lr-card-details">
        <div className="lr-card-title">
          <Link
            to={`/${props.page}/${item.id}`}
            className="heading-starwars text-highlight "
          >
            {item.name}
          </Link>
        </div>
      </div>
    </div>
  );
};
