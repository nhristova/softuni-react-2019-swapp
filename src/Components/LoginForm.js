import React, { useState } from 'react';

export const LoginForm = props => {
  let [emailVal, setEmail] = useState('');
  let [passwordVal, setPassword] = useState('');

  const handleEmailChange = ({ target: { value: email } }) => {
    setEmail(email);
  };

  const handlePasswordChange = ({ target: { value: pass } }) => {
    setPassword(pass);
  };

  const handleSubmit = e => {
    e.preventDefault();
    props.signIn({ variables: { email: emailVal, password: passwordVal } });
  };

  return (
    <div className="login-modal">
      <h2 className="login-heading" onClick={props.toggleTheme}>
        SWAPP
      </h2>
      <form className="login-card" onSubmit={handleSubmit}>
        <div
          id="login-error"
          className={`login-error-message ${props.showError ? '' : 'hidden'}`}
        >
          Something went wrong with signin
        </div>
        <input
          className="login-input"
          type="email"
          placeholder="email"
          onChange={handleEmailChange}
          value={emailVal}
        />
        <input
          className="login-input"
          type="password"
          placeholder="password"
          onChange={handlePasswordChange}
          value={passwordVal}
        />
        <button className="load-more-button" type="submit">
          Login
        </button>
      </form>
    </div>
  );
};
