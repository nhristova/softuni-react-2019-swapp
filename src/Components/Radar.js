import React from 'react';
import gql from 'graphql-tag.macro';
import RadarChart from 'react-svg-radar-chart';
import { useQuery } from '@apollo/react-hooks';

export const STARSHIP_CLASS_QUERY = gql`
  query($shipType: String) {
    allStarships(first: 100, after: "0", filter: { starshipClass: $shipType }) {
      edges {
        node {
          id
          starshipClass
          cost
          maxAtmosphericSpeed
          maxMLPerHour
          hyperdriveRating
          crew
        }
      }
    }
  }
`;

const getMax = (first, second) => {
  if (first > second) {
    return first;
  }
  return second;
};

const maxValues = {
  cost: 0,
  maxAtmosphericSpeed: 0,
  maxMLPerHour: 0,
  hyperdriveRating: 0,
  crew: 0,
};

const chartCaptions = {
  cost: 'Cost',
  maxAtmosphericSpeed: 'Max Atm. Speed',
  maxMLPerHour: 'Max ML/h',
  hyperdriveRating: 'HyperD Rat.',
  crew: 'Crew',
};

const options = {
  size: 400,
  scales: 5,
  axes: false,
  dots: true,
};

export const Radar = props => {
  const starship = props.starship;
  const { data, loading, error } = useQuery(STARSHIP_CLASS_QUERY, {
    variables: { type: starship.starshipClass },
  });

  if (loading) return <div>Loading</div>;
  if (error) return <p>Error</p>;

  const { allStarships } = data;

  allStarships.edges.forEach(el => {
    maxValues.cost = getMax(el.node.cost, maxValues.cost);
    maxValues.maxAtmosphericSpeed = getMax(
      el.node.maxAtmosphericSpeed,
      maxValues.maxAtmosphericSpeed,
    );
    maxValues.maxMLPerHour = getMax(
      el.node.maxMLPerHour,
      maxValues.maxMLPerHour,
    );
    maxValues.hyperdriveRating = getMax(
      el.node.hyperdriveRating,
      maxValues.hyperdriveRating,
    );
    maxValues.crew = getMax(el.node.crew, maxValues.crew);
  });

  const chartData = [
    {
      data: {
        cost: starship.cost / maxValues.cost,
        maxAtmosphericSpeed:
          starship.maxAtmosphericSpeed / maxValues.maxAtmosphericSpeed,
        maxMLPerHour: starship.maxMLPerHour / maxValues.maxMLPerHour,
        hyperdriveRating:
          starship.hyperdriveRating / maxValues.hyperdriveRating,
        crew: starship.crew / maxValues.crew,
      },
      meta: { color: '#4bd5ee' },
    },
  ];

  return (
    <div className="starship-radar-chart">
      <RadarChart
        captions={chartCaptions}
        data={chartData}
        size={400}
        options={options}
      />
    </div>
  );
};
