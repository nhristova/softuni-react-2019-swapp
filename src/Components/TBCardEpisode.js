import React from 'react';
import { Link } from 'react-router-dom';

export const TBCardEpisode = prop => {
  const episode = prop.episode;
  return (
    <div className="episode-card tb-card">
      <div className="tb-card-photo">
        <img src={episode.image} alt={`Poster ${episode.title}`} />
      </div>
      <div className="tb-card-title">
        <Link
          to={`/episodes/${episode.id}`}
          className="heading-starwars text-highlight"
        >
          {episode.title}
        </Link>
      </div>
      <div className="tb-card-description">{episode.openingCrawl}</div>
    </div>
  );
};
