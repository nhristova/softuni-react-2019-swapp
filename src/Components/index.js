export * from './Login';
export * from './Episodes';
export * from './pages';
