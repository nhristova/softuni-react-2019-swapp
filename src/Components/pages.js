import React from 'react';
import { Switch, Redirect, Route } from 'react-router-dom';
import { Header } from './Header';
import { Episodes } from './Episodes';
import { Characters } from './Characters';

export const Pages = props => {
  return (
    <>
      <Header changeTheme={props.toggleTheme} />
      <Switch>
        <Route path="/episodes" component={Episodes} />
        <Route path="/characters" component={Characters} />
        <Redirect from="/login" to="/episodes" />
        <Redirect from="/" to="/episodes" />
      </Switch>
    </>
  );
};
