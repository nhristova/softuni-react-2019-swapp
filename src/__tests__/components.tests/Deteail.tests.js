import React from 'react';
import renderer from 'react-test-renderer';
import { Detail } from '../../Components';

describe('Detail component', () => {
  it('renders correctly', () => {
    const cmp = renderer.create(<Detail type="test type" value="test value" />);
    expect(cmp).toMatchSnapshot();
  });
});
