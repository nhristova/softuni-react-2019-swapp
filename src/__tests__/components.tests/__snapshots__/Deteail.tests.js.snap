// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component renders correctly 1`] = `
<div>
  <span
    className="text-highlight-2"
  >
    test type
    : 
  </span>
  <span
    className="text-highlight"
  >
    test value
  </span>
</div>
`;
